tests/Cargo.toml

```diff
@@ -16,6 +16,11 @@ name = "wgpu-test"
 path = "tests/root.rs"
 harness = false
 
+[[test]]
+name = "wgpu-compile-test"
+path = "compile_tests/root.rs"
+harness = true
```

Comment:
Lookie: Our test suite has grown a new feature. We can now ensure that a given bit of wgpu code _does not_ compile. This is very useful to make sure that #6145 doesn't come back.